docs/adr/0002-stable-tenant-ids.md

```diff
@@ -0,0 +1,25 @@
+# Support for stable tenant Ids
+
+## Status
+
+Accepted
+
+## Context
+
+Up until now, tenant Ids have always been automatically generated. As we have used Corvus.Tenancy as part of Marain, we have come across scenarios where it is useful to control what a tenant's Id will be. This primarily applies to the tenants we use as top-level containers for Client and Service tenants, as well as the service tenants themselves.
+
+In the absence of the ability to know tenant Ids in advance we have fallen back on well known names for these tenants, effectively introducing an alternative well-known Id for them. As well as introducing a potential point of failure due to the need to keep these names unique, it is far less efficient to locate a tenant by name than by Id.
+
+## Decision
+
+To avoid the need to effectively introduce an alternative way of identifying tenants, we will allow the Id of tenants to be controlled when they are created.
+
+This will be done by adding a new method to the `ITenantProvider` interface: `CreateWellKnownChildTenantAsync`. This allows callers to specify a GUID which is used to generate the new tenant's ID, rather than generating a random GUID internally (which is what the existing `CreateChildTenantAsync` method does).
+
+The resulting Id of the new tenant will still be generated by concatenating the parent tenant's Id with the hash of the provided GUID. This means that in order for a tenant to have a well known Id, all of its ancestors must also have well-known Ids.
+
+In order to prevent two tenants being created with the same Id, it is necessary to check that the Id is not already in use prior to creating the new tenant.
+
+## Consequences
+
+Allowing clients to control tenant Ids means that we can no longer guarantee they are globally unique - i.e. unique across different tenant providers. Should there ever be a need to migrate tenants between stores, this change will make the process more complex.
```

Comment:
I'm confused by this last paragraph. It seems to me like it makes migration easier. The ability to control IDs means we can now copy a tenant into a new environment without changing its ID.